saleor/attribute/migrations/tasks/saleor3_16.py

```diff
@@ -0,0 +1,57 @@
+from ....celeryconf import app
+from ...models import AssignedPageAttributeValue
+
+from django.db import transaction, connection
+
+# batch size to make sure that task is completed in 1 second and
+# as well we don't use too much memory
+BATCH_SIZE = 1000
+
+
+def update_page_assignment():
+    """Update Page assignment.
+
+    Update a batch of 'AssignedPageAttributeValue' rows by setting their 'page' based
+    on their related 'assignment'.
+
+    The number of rows updated in each batch is determined by the BATCH_SIZE.
+    Rows are locked during the update to prevent concurrent modifications.
+    """
+    with transaction.atomic():
+        with connection.cursor() as cursor:
+            cursor.execute(
```

Comment:
Have you tested the update logic with DB that has +/- million attributes?

Reply:
Yes and single task range for 1_000_000 rows was min/max 0.15/0.52 second so based on that I have increased `BATCH_SIZE` to 5000. 
It was tested on my local machine.